File: manual_transpilation/gperf_rust/src/bool_array.rs
pub struct Bool_Array
{
    pub _size: u32,
    pub _iteration_number: u32,
    pub _storage_array: Vec<u32>,
}

impl Bool_Array {
    pub fn new(size: u32) -> Bool_Array {
        Bool_Array{
            _size : size,
            _iteration_number : 0,
            _storage_array : vec![0; size as usize],
        }

    }

    #[inline]
    pub fn set_bit(&mut self, index: u32) -> bool {
        if self._storage_array[index as usize] == self._iteration_number
        {
            return true;
        }
        else
        {
            self._storage_array[index as usize] = self._iteration_number;
            return false;
        }
    }

    #[inline]
    pub fn clear(&mut self) {
        self._iteration_number += 1;

        if self._iteration_number == 0
        {
            self._iteration_number = 1;
            self._storage_array = vec![0; self._size as usize];
        }
    }
}

fn main() {
    
}

File: TranspilerExecutables/ANTLRVersion_Examples/Example_Conversions/BoolArray.cpp
#include <bits/stdc++.h>
using namespace std;

class Bool_Array
{
public:
    Bool_Array(unsigned int size) : _size(size),
                                    _iteration_number(1),
                                    _storage_array(new unsigned int[size])
    {
        memset(_storage_array, 0, size * sizeof(_storage_array[0]));
        if (option[DEBUG])
            fprintf(stderr, "\nbool array size = %d, total bytes = %d\n",
                    _size,
                    static_cast<unsigned int>(_size * sizeof(_storage_array[0])));
    }

    void clear()
    {

        if (++_iteration_number == 0)
        {
            _iteration_number = 1;
            memset(_storage_array, 0, _size * sizeof(_storage_array[0]));
            if (option[DEBUG])
            {
                fprintf(stderr, "(re-initialized bool_array)\n");
                fflush(stderr);
            }
        }
    }

    bool set_bit(unsigned int index)
    {
        if (_storage_array[index] == _iteration_number)

            return true;
        else
        {

            _storage_array[index] = _iteration_number;
            return false;
        }
    }

private:
    unsigned int const _size;
    unsigned int _iteration_number;
    unsigned int *const _storage_array;
};
